lms/static/scripts/frontend_apps/components/AutoGradingConfigurator.tsx

```diff
@@ -0,0 +1,208 @@
+import {
+  Checkbox,
+  CheckboxCheckedFilledIcon,
+  Input,
+  RadioGroup,
+} from '@hypothesis/frontend-shared';
+import type { ComponentChildren } from 'preact';
+import { useCallback, useId } from 'preact/hooks';
+
+export type GradingType = 'all_or_nothing' | 'scaled';
+
+export type AutoGradingConfig = {
+  /** Whether auto grading is active for the assignment or not */
+  active?: boolean;
+
+  /**
+   * - all_or_nothing: students need to meet a minimum value, making them get
+   *                   either 0% or 100%
+   * - scaled: students may get a proportional grade based on the amount of
+   *           annotations. If requirement is 4, and they created 3, they'll
+   *           get a 75%
+   */
+  gradingType: GradingType;
+
+  /**
+   * - cumulative: both annotations and replies will be counted together for
+   *               the grade calculation
+   * - separately: students will have different annotation and reply goals.
+   */
+  activityCalculation: 'cumulative' | 'separately';
```

Comment:
You could perhaps model this with something like:

```ts
type ActivityRequirement = {
  type: "cumulative",
  annotations: number,
} | {
  type: "separate",
  annotations: number,
  replies: number,
}
```

Reply:
I though about that, but it required extra type checks in order to read and set the data.

I'll give it another thought, but in this case it seemed more straightforward to just set optional props and let the correct shape be driven by the UI.

Reply:
This data shape is going to be shared between BE/FE so whatever works well in that context works for me.

Reply:
The suggestion there should not affect how the data is exchanged with the server, just how it is used locally.